Agents/ForecastingAgent/forecasting/utils/useful_queries.py

```diff
@@ -0,0 +1,200 @@
+from forecasting.datamodel.iris import *
+from forecasting.utils.tools import *
+import pandas as pd
+from darts import concatenate
+
+
+def get_properties_for_subj(subj: str, verb_obj: dict = {}, verb_literal: dict = {}):
+    """
+         Constructs proper SPARQL update string from given name, verb-object and verb-literal dictionary.
+         Arguments:
+              subj: subject
+              verb_obj:  object properties: dictionary of verb-object pairs, if value is a list, then it will be treated as multiple objects for the same verb
+              verb_literal: data properties: dictionary of verb-literal pairs, if value is a list, then it will be treated as a literal with type
+         Returns:
+              SPARQL update string
+    """
+    output = f''''''
+    #  object properties
+    for verb, obj in verb_obj.items():
+        # turn obj to list if it is not to iterate over it
+        if not isinstance(obj, list):
+            obj = [obj]
+        for o in obj:
+            output += f'''{create_triple(subj, verb, o)}'''
+
+    # data properties
+    for verb, literal in verb_literal.items():
+        if not isinstance(literal, list):
+            literal = [literal]
+        if len(literal) == 1:
+            literal.append(None)
+        # check if len is max 2 for [literal, type]
+        if len(literal) > 2:
+            raise ValueError(
+                f'Literal {literal} has more than two elements expected [literal, type] or [literal]')
+        if not isinstance(literal[0], str):
+            literal[0] = str(literal[0])
+        # check that literal does not start with < and end with >
+        if literal[0].startswith('<') and literal[0].endswith('>'):
+            raise ValueError(f'Literal {literal} starts with < or ends with >')
+        output += f'''{create_triple_literal(subj, verb, literal[0], literal[1])}'''
+
+    return output
+
+
+def get_predecessor_type_and_predicate(iri, kgClient):
+    # get predecessor type of iri
+    query = f"""
+    SELECT ?predecessorType ?predicate
+    WHERE {{
+        ?o ?predicate <{iri}> . 
+        ?o <{RDF_TYPE}> ?predecessorType . 
+        }}"""
+    cfg = kgClient.performQuery(query)
+    return {r["predicate"]: r["predecessorType"] for r in cfg}
+
+
+def get_ts_value_iri(iri, kgClient):
+    # get ts value iri of iri
+    query = f"""
+    SELECT ?tsValueIRI
+    WHERE {{
+        <{iri}> <{OM_HASVALUE}> ?tsValueIRI . 
+        }}"""
+    ts_value_iri = kgClient.performQuery(query)[0]["tsValueIRI"]
+    return ts_value_iri
+
+
+def check_cov_matches_rdf_type(row, rdf_type):
+    if 'type_with_measure' in row and row['type_with_measure'] == rdf_type or 'type_without_measure' in row and row['type_without_measure'] == rdf_type:
+        return True
+    else:
+        return False
+
+
+def get_covs_heat_supply(kgClient, tsClient,  lowerbound, upperbound, df=None):
```

Comment:
never used

Reply:
but can be useful to retrieve its time index to calculate time covariates, if no other covariates are given to